wordpress-mobile/WordPress-iOS
# Likes Notification: Display likes using the new endpoints

Refs #15662 

Since the likes data will be fetched from a service, `LikesListController` is introduced to encapsulate the fetching logic. The component mainly serves as a replacement for the table view's delegate and data source in `NotificationDetailsViewController`. Here are a couple of reasons why I chose this approach:

- Minimal impact on the current implementation.
- Minimal effort to implement feature flag (see `setupTableDelegates` method in `NotificationDetailsViewController`).
- Makes it easy to modify later on once we want to implement the Likes page from Reader (e.g. create a thin view controller that uses this component).

By checking out this PR, you should be able to see the look and feel of the new changes directly. However, there are some areas that are not yet covered in this PR:
- [x] Implement feature flag.
- [ ] Add tests for the `LikesListController`. In order to minimize the PR size, I plan to separate the test into another PR.
- [ ] Ensure that the changes do not impact accessibility.
- [ ] Implement Tracks.

Additionally, some review comments from previous PRs were addressed in this change:
- https://github.com/wordpress-mobile/WordPress-iOS/pull/15977#discussion_r585114909
- https://github.com/wordpress-mobile/WordPress-iOS/pull/15994#discussion_r585122810

To test:

- Ensure that the app builds and that you are logged in using an account that already has existing notifications.
- Go to the Notifications tab, and select a notification that's either post likes or comment likes – preferably one that has more than 10 likes. Verify that the list displays all users that liked the content.
- Try to navigate between notifications through the previous/next buttons on the top right navigation bar, and verify that the contents are displayed properly.

PR submission checklist:
